### PR TITLE
Support speech playback from memory

### PR DESCRIPTION
**Pull request by p-robertson**
to reduce drive fatigue from write/read/erase
operations.

    - changed from playback library
        to pydub library
    - support brian, eleven labs, and gtts

	modified:   autogpt/speech/brian.py
	modified:   autogpt/speech/eleven_labs.py
	modified:   autogpt/speech/gtts.py
	new file:   autogpt/speech/playback.py

### Background

Playing audio from memory can be faster and more efficient than writing it to a file on the drive and then reading it back. This is because accessing data from memory is generally faster than accessing data from a drive. Additionally, writing and deleting files frequently can cause wear and tear on solid-state drives (SSDs), reducing their lifespan.

### Changes

I have added a new file playback.py under the autogpt/speech directory. This file contains the play_from_memory function which allows for audio playback directly from memory in brian.py, eleven_labs.py, and gts.py within the autogpt/speech directory. This eliminates the need for download/write/read/delete operations. Additionally, I have replaced the playsound library with pydub for audio playback. The play_from_memory function uses the multiprocessing library to avoid console output from the pydub library.

### Test Plan

I have manually and thoroughly tested the changes introduced in this pull request. Additionally, I have run the included tests to ensure that everything is working as expected. Please note that the tests file needs to be updated to change the start directory from autogpt/tests to tests.

### PR Quality Checklist
- [/] My pull request is atomic and focuses on a single change.
- [/] I have thoroughly tested my changes with multiple different prompts.
- [/] I have considered potential risks and mitigations for my changes.
- [/] I have documented my changes clearly and comprehensively.
- [/] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — DISMISSED** [review]

**Comment by nponeccop** [comment]
@p-robertson CI is red

**Comment by p-robertson** [comment]
This requires pydub, need to update requirements.txt  Also looks like you're building with 3.8, so I can also convert the | union to a typing Union and add docstrings to the playback_from_memory function.

**Comment by p-robertson** [comment]
requirements.txt has been updated for pydub, tests.py has been synced, the pipe union has been replaced with a Union[] and docstrings have been added to the playback_from_memory function.

File: tests.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 if __name__ == "__main__":
     # Load all tests from the 'autogpt/tests' package
-    suite = unittest.defaultTestLoader.discover("autogpt/tests")
+    suite = unittest.defaultTestLoader.discover("./tests")
```
Comment (nponeccop) [inline tests.py]: Is this change related?

**Review by nponeccop — APPROVED** [review]

**Comment by p-robertson** [comment]
Many apologies, it would appear I somehow accidentally resynced with the main project and this was automatically closed as a result.  Hopefully you can salvage the code in here, or at your request I can reimplement it into a new branch and do another pull request.

**Comment by p-robertson** [comment]
I am going to rewrite using pygame's mixer instead of pydub since pydub requires ffmpeg or libav [abadonware] to be installed on the system.
